src/libfsm/exec.c

```diff
@@ -53,7 +53,7 @@ fsm_exec(const struct fsm *fsm,
 	assert(fsm->start != NULL);
 	state = fsm->start;
 
-	while (c = fsm_getc(opaque), c != EOF) {
+	while (c = fsm_getc(opaque), c != (unsigned int)EOF) {
```

Comment:
`c` itself should be declared `int c;` and no casts here. Just like when using the standard `getc()`.